panel/tests/ui/layout/test_feed.py

```diff
@@ -21,27 +21,17 @@ def test_feed_load_entries(page):
 
     expect(feed_el).to_have_class("bk-panel-models-feed-Feed scroll-vertical")
 
-    children_count = feed_el.evaluate(
-        '(element) => element.shadowRoot.querySelectorAll(".bk-panel-models-markup-HTML").length'
-    )
+    children_count = feed_el.locator('.bk-panel-models-markup-HTML').count()
     assert 50 <= children_count <= 65
 
     # Now scroll to somewhere down
     feed_el.evaluate('(el) => el.scrollTo({top: 100})')
-    children_count = feed_el.evaluate(
-        '(element) => element.shadowRoot.querySelectorAll(".bk-panel-models-markup-HTML").length'
-    )
+    children_count = feed_el.locator('.bk-panel-models-markup-HTML').count()
     assert 50 <= children_count <= 65
 
     # Now scroll to top
     feed_el.evaluate('(el) => el.scrollTo({top: 0})')
-    wait_until(
-        lambda: feed_el.evaluate(
-            '(element) => element.shadowRoot.querySelectorAll(".bk-panel-models-markup-HTML").length'
-        )
-        >= 50
-    )
-
+    wait_until(lambda: feed_el.locator('.bk-panel-models-markup-HTML').count() >= 50, page)
```

Comment:
@ahuang11 Quick not for future reference, `evaluate` should only ever be needed for actual JS evaluation and `locator` is always preferable over `query_selector`.